### DISCUSSION TITLE
PrimeReactProvider for multiple roots

### DISCUSSION THREAD

**Discussion by leqwasd** [primefaces/community#1048] [PrimeReact]
Hello,

Our app is not 100% react. Whenever we need a react components - we "ReactDOM.render" into an element and move on. Sometimes this is a whole module / page. Sometimes it is as little as a single input element.
This results in multiple "roots" from the perspective of react.

I am now struggling to update prime-react to v10. It has to have `PrimeReactProvider` wrapped around the app. Because we have multiple roots - I don't really know - how to do this reasonably?

Wrapping every "root" render into new PrimeReactProvider? Seems pretty wasteful.

I was wondering - If there would be an API - to call and initialize once this context without using react context and rendering - that would make this work easier :)

**Comment by melloware** [comment]
A lot of people recommend using useContextBridge: https://github.com/pmndrs/its-fine?tab=readme-ov-file#usecontextbridge

See: https://stackoverflow.com/questions/77880918/making-react-context-available-without-top-level-wrapping
